### DISCUSSION TITLE
Question: How to produce key-pair to use with SSH server?

### DISCUSSION THREAD

**Discussion by bushev** [paulmillr/noble-ed25519#36] [General]
Hi,

I'm looking at how to use this library to generate SSH key-pairs for Linux SSH servers. I have a question. Can this library be used to produce keys as this command does? 

```
ssh-keygen -o -a 100 -t ed25519 -f id_ed25519
```
I tried to use hex key-pair from this library to test here (http://ed25519.herokuapp.com/), but it doesn't verify my signature after signing.

Thanks!

**Comment by paulmillr** [comment]
SSH encoding is a bit complex. It requires specific padding scheme and structures. I'm not aware of popular repos that have it implemented in js.

**Comment by aral** [comment]
In case anyone else stumbles on this, @paulmillr’s own https://github.com/paulmillr/ed25519-keygen library now allows this.

**Reply by paulmillr** [reply]
Exactly. Thanks Aral.

**Reply by aral** [reply]
Thank you, Paul. Your work is going straight into the heart of the Small Web stuff I’m building (https://ar.al/2020/08/07/what-is-the-small-web/) and has saved me goodness knows how much time (and how many bytes) :)
